test/e2e/compatibilities_test.go

```diff
@@ -24,41 +25,54 @@ func TestKongRouterFlavorCompatibility(t *testing.T) {
 
 	t.Log("deploying kong components with traditional Kong router")
 	manifest := getTestManifest(t, dblessPath)
-	_ = deployKong(ctx, t, env, manifest)
-	ensureGatewayDeployedWithRouterFlavor(ctx, t, env, "traditional")
+	deployKong(ctx, t, env, manifest)
+	proxyDeploymentNN := getManifestDeployments(dblessPath).ProxyNN
+	ensureGatewayDeployedWithRouterFlavor(ctx, t, env, proxyDeploymentNN, "traditional")
 
 	t.Log("running ingress tests to verify that KIC with traditonal Kong router works")
 	deployIngress(ctx, t, env)
 	verifyIngress(ctx, t, env)
 
-	setGatewayRouterFlavor(ctx, t, cluster, "traditional_compatible")
+	setGatewayRouterFlavor(ctx, t, cluster, proxyDeploymentNN, "traditional_compatible")
 
 	t.Log("waiting for Kong with traditional_compatible router to start")
-	ensureGatewayDeployedWithRouterFlavor(ctx, t, env, "traditional_compatible")
+	ensureGatewayDeployedWithRouterFlavor(ctx, t, env, proxyDeploymentNN, "traditional_compatible")
 
 	t.Log("running ingress tests to verify that KIC with traditonal_compatible Kong router works")
 	verifyIngress(ctx, t, env)
 }
 
-func setGatewayRouterFlavor(ctx context.Context, t *testing.T, cluster clusters.Cluster, flavor string) {
+func setGatewayRouterFlavor(
+	ctx context.Context,
+	t *testing.T,
+	cluster clusters.Cluster,
+	proxyDeploymentNN types.NamespacedName,
+	flavor string,
+) {
 	// Since we cannot replace env vars in kustomize, here we update the deployment to set KONG_ROUTER_FLAVOR to traditional_compatible.
 	t.Log("update deployment to modify Kong's router to traditional_compatible")
-	gatewayDeployment, err := cluster.Client().AppsV1().Deployments(namespace).Get(ctx, "proxy-kong", metav1.GetOptions{})
+	gatewayDeployment, err := cluster.Client().AppsV1().Deployments(proxyDeploymentNN.Namespace).Get(ctx, proxyDeploymentNN.Name, metav1.GetOptions{})
```

Comment:
nit: `cluster.Client().AppsV1().Deployments(proxyDeploymentNN.Namespace)` can also be extracted to a variable.